source/controllers/maintenance.py

```diff
@@ -37,33 +37,36 @@ class MaintenanceController(object):
     def get_services():
         """
         Retrieve all configured maintenance service running on this node for each backend
-        :return: generator
+        :return: All maintenance services for each ALBA Backend
+        :rtype: generator
         """
         for service_name in MaintenanceController._service_manager.list_services(MaintenanceController._local_client):
             if service_name.startswith(MaintenanceController.MAINTENANCE_PREFIX):
                 yield service_name
 
     @staticmethod
-    def add_maintenance_service(name, backend_guid, abm_name):
+    def add_maintenance_service(name, alba_backend_guid, abm_name, read_preferences=None):
         """
         Add a maintenance service with a specific name
         :param name: Name of the service to add
         :type name: str
-        :param backend_guid: Backend for which the maintenance service needs to run
-        :type backend_guid: str
+        :param alba_backend_guid: ALBA Backend GUID for which the maintenance service needs to run
+        :type alba_backend_guid: str
         :param abm_name: Name of the ABM cluster
         :type abm_name: str
+        :param read_preferences: List of ALBA Node IDs (LOCAL) or ALBA IDs of linked ALBA Backends (GLOBAL) for the maintenance services where they should prioritize the READ actions
+        :type read_preferences: list[str]
+        :return: None
+        :rtype: NoneType
         """
         if MaintenanceController._service_manager.has_service(name, MaintenanceController._local_client) is False:
-            config_location = '/ovs/alba/backends/{0}/maintenance/config'.format(backend_guid)
-            alba_config = Configuration.get_configuration_path(config_location)
-            node_id = SettingList.get_setting_by_code(code='node_id').value
-            params = {'ALBA_CONFIG': alba_config,
+            config_location = '{0}/config'.format(MaintenanceController.MAINTENANCE_KEY.format(alba_backend_guid, name))
```

Comment:
Every new  maintenance gets a new config.
What about existing maintenances? :thinking:

Reply:
That's where the migration code comes in